math/overflow.go

```diff
@@ -29,6 +29,27 @@ func AddUint64WithErr(a, b uint64) (uint64, error) {
 	return s, ErrAdditionOverflow
 }
 
+// AddInt64 performs addition on uint64 and logs an error if the addition overflows
+func AddInt64(a, b int64) int64 {
+	res, err := AddInt64WithErr(a, b)
+	if err != nil {
+		log.Trace("AddUint64 overflow", "a", a, "b", b)
+		return builtinMath.MaxInt64
+	}
+
+	return res
+}
+
+// AddInt64WithErr performs addition on int64 and returns an error if the addition overflows
+func AddInt64WithErr(a, b int64) (int64, error) {
+	s := a + b
+	if (s > a) == (b > 0) {
+		return s, nil
+	}
+
+	return s, ErrAdditionOverflow
```

Comment:
Fixed everywhere: the correct value to return in case of failure is the maximum value corresponding to the return type (e.g. `MaxUint64`). Yes, Go's convention is to return the default value in case of error, but that's dangerous here, because a human error could have nasty consequences.